### ISSUE TITLE
Importing `geopandas` prevents `panel` from rendering `servable()` components

### ISSUE THREAD

**Issue by alaws-USGS** [nebari-dev/nebari#2562]
### Describe the bug

When a file, either .py or .ipynb, has the following line in it, `import geopandas as gpd`, it gives the following error when trying to preview the Panel app:
```
Error Message: The Panel application being served did not serve any contents. Ensure you mark one or more Panel components in your notebook or script with .servable(), e.g. pn.Row('Hello World!').servable().
```
It also does not render when you create an app and start an instance from the landing page.

Testing with another app not using geopandas allows it to render.

### Expected behavior

The Panel app to render.

### OS and architecture in which you are running Nebari

Linux/AWS EC2

### How to Reproduce the problem?

```python
import geopandas as gpd
import panel as pn

pn.Row(pn.pane.Markdown("Some text")).servable()
```

### Versions and dependencies used.

channels:
  - conda-forge
dependencies:
  - python=3.11
  - adlfs
  - aiobotocore
  - awscli
  - birdy
  - black
  - bokeh
  - boto3
  - cdsapi
  - cf_xarray
  - cfgrib
  - cfunits
  - climpred
  - coiled
  - curl
  - dask-geopandas
  - dask
  - dask-gateway==2023.9.0
  - datashader
  - datacube
  - dask-geopandas
  - depfinder
  - distributed
  - erddapy
  - fastparquet
  - flox
  - folium
  - fsspec>=2024.2.0
  - gcsfs
  - gdal
  - gdptools
  - geocube
  - geogif
  - geolinks
  - geopandas
  - geopy
  - geoviews
  - graphviz
  - h5netcdf
  - h5py
  - hologridgen
  - holoviews>=1.18.3
  - htop
  - hvplot
  - imagecodecs
  - intake>=2.0
  - intake-geopandas
  - intake-stac
  - intake-xarray
  - intake-parquet
  - ipykernel
  - ipyleaflet
  - ipywidgets
  - isort
  - jinja2
  - jupyter_bokeh
  - jupyter-panel-proxy
  - jupyterlab_code_formatter
  - jupytext
  - jq
  - kerchunk
  - leafmap
  - lxml
  - lz4
  - mamba
  - metpy
  - nbgitpuller
  - nbstripout
  - nco
  - netcdf4 == 1.6.0
  - numba
  - numcodecs
  - odc-algo
  - odc-stac
  - openpyxl
  - osmnx
  - owslib
  - pandas == 1.5.3
  - panel
  - pangeo-forge-recipes
  - papermill
  - param
  - pip
  - pint-xarray
  - planetary-computer
  - pyarrow
  - pyepsg
  - pygeohydro
  - pydaymet
  - pynhd
  - py3dep
  - pygeoogc
  - pygeoutils
  - pynco
  - pyogrio
  - pyvista
  - async_retriever
  - pystac
  - pystac-client
  - python-graphviz
  - python-snappy
  - pyyaml
  - python-gist
  - rasterio
  - rechunker
  - requests
  - rich
  - rio-cogeo
  - rioxarray
  - scikit-image
  - s3fs>=2023.2.0
  - seawater
  - selenium
  - siphon
  - spatialpandas
  - stackstac
  - toolz
  - ujson
  - unzip
  - utide
  - vim
  - wgrib2
  - xagg
  - xarray-spatial
  - xarray_leaflet
  - xbitinfo-python
  - xesmf
  - xoak
  - xrviz
  - xskillscore
  - zip
  - zstandard
  - xmip
  - intake-esm
  - xarray-datatree
  - h5pyd
  - noaa-coops
  - kbatch
  - xstac
  - zarr
  - jupyter-book
  - ghp-import
  - jsonschema-with-format-nongpl
  - webcolors
  - xarrayutils
  - xroms
  - jhsingle-native-proxy
  - bokeh-root-cmd
variables: {}

### Compute environment

AWS

**Comment by Adam-D-Lewis** [comment]
Hi @alaws-USGS, thanks for opening an issue.  I'm curious if you've tried to reproduce this locally (not on Nebari).  If you are able to reproduce locally, then it suggests the issue is not Nebari specific and the fix may need to be with panel or geopandas or one of the other dependencies.

**Comment by alaws-USGS** [comment]
Hi @Adam-D-Lewis, I have deployed the same app locally and have not been able to replicate the same issue. I also perused both the `panel` and `geopandas` repos and not found a mention of this issue.

**Comment by Adam-D-Lewis** [comment]
> Hi @Adam-D-Lewis, I have deployed the same app locally and have not been able to replicate the same issue. I also perused both the `panel` and `geopandas` repos and not found a mention of this issue.

Thank you @alaws-USGS.  I have seen some problems with visualization libraries used on Nebari.  More info is [here](https://github.com/nebari-dev/nebari/issues/2261).  I was able to resolve by pinning several dependencies (ipympl, jupyterlab, jupyterlab_widgets, jupyterlab_server, jupyter_client, ipykernel) to match up with Nebari's jupyterlab image.  Be aware that the versions pinned in that issue are no longer the correct versions in the latest version of Nebari.  I'm not sure this is the solution, but would be the first thing I'd recommend to try.

**Comment by alaws-USGS** [comment]
HI @Adam-D-Lewis, I read your suggestion for the fix and thank you for sharing that. I believe we are using a version of Nebari that matches one of those two so it should hopefully be seamless! I will also report back on what the results are.

I want to echo what you said in your issue and say that this is not something a Nebari user should have to hunt down and manually adjust. I believe that it needs to be addressed in order to make Nebari production-ready.

**Comment by Adam-D-Lewis** [comment]
> I want to echo what you said in your issue and say that this is not something a Nebari user should have to hunt down and manually adjust. I believe that it needs to be addressed in order to make Nebari production-ready.

I agree it's not ideal.  There are a few potential solutions I've learned about recently that I would like to test soon - https://github.com/nebari-dev/nebari/issues/2261#issuecomment-2231346756.

**Comment by kcpevey** [comment]
You mentioned "Panel Preview" so I'm wondering if you are seeing this error when viewing through the Jupyter Notebook "Panel Preview" button? Does it render in a notebook cell?

I have had issues with the Panel Preview button on Nebari. I never tracked down the issue - it was working for me at one point, then it stopped working on my project. I haven't gone back to test with other projects/environments yet. 

Also, I dont think this is related, but just in case - I've had some issues [with cache when using Panel on Nebari](https://github.com/nebari-dev/nebari-docs/issues/464) that cause the render to fail in the notebook. You might want to check that out?

**Comment by kcpevey** [comment]
I noticed you didn't load the panel extension in the notebook. You will definitely need to add that. 

I tried building your environment but its very heavy and is still building after 45 minutes, so I made a smaller one for testing:

```
channels:
  - conda-forge
dependencies:
  - python=3.10
  - geopandas
  - panel
  - param
  - ipykernel
 ```

In a Notebook, if I add the `pn.extension()`, it will render

![image](https://github.com/user-attachments/assets/28668301-20ed-4991-8d80-073f44a7fef9)


I have noticed a race condition in the past where the `pn.extension()` call must be made in a separate cell. Not sure if that's still an issue, but just something to be aware of. 

All that said, I tried to the `Panel Preview` button in the notebook and that did NOT work. The preview pane itself is showing `nebari.quansight.dev refused to connect.` and the js console is showing the error: `Refused to frame 'https://nebari.quansight.dev/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'none'".`

I removed the geopandas import and I'm seeing the same thing. I think this issue boils down to Panel Preview not working on Nebari, which looks like it may be a security issue within Jupyter.

**Comment by krassowski** [comment]
JupyterHub 4.1 introduced pretty restrictive CSP. In principle we can change CSP with `c.JupyterHub.tornado_settings` traitlet.

For reference:
- https://jupyter-server.readthedocs.io/en/latest/operators/public-server.html#content-security-policy-csp
- https://jupyterhub.readthedocs.io/en/stable/explanation/websecurity.html#mitigating-same-origin-deployments

**Comment by krassowski** [comment]
Here is a related upstream issue:
- https://github.com/jupyterhub/jupyterhub/issues/4823

Quoting:

> This is required for inter-user security on single-domain deployments, but can be safely disabled with per-user subdomains.

Nebari should implement per-user domains and then all these restrictions can be safely lifted.

**Comment by kcpevey** [comment]
I have moved the issue of Panel Preview being unavailable to its [own issue](https://github.com/nebari-dev/nebari/issues/2580).
